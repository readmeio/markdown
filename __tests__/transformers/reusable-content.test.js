import { mdast } from '../../index';

describe('reusable content transfomer', () => {
  it('should replace a reusable content block if the block is provided', () => {
    const reusableContent = {
      Test: `
# Test

[link](http://example.com)
    `,
    };
    const md = `
Before

<Test />

After
    `;

    const tree = mdast(md, { reusableContent });

    expect(tree.children[0].children[0].value).toBe('Before');
    expect(tree.children[1]).toMatchSnapshot();
    expect(tree.children[2].children[0].value).toBe('After');
  });

  it('should replace a reusable content block with multiple words if the block is provided', () => {
    const reusableContent = {
      MyCustomComponent: `
# Test

[link](http://example.com)
    `,
    };
    const md = '<MyCustomComponent />';

    const tree = mdast(md, { reusableContent });

    expect(tree.children[0]).toMatchSnapshot();
  });

  it('should insert an empty node if the reusable content block is not defined', () => {
    const md = '<NotDefined />';
    const tree = mdast(md);

    expect(tree.children[0].type).toBe('reusable-content');
    expect(tree.children[0].children).toStrictEqual([]);
  });

  it('does not expand reusable content recursively', () => {
    const reusableContent = {
      Test: '<Test />',
    };
    const md = '<Test />';
    const tree = mdast(md, { reusableContent });

    expect(tree.children[0].children[0].type).toBe('reusable-content');
    expect(tree.children[0].children[0].children).toStrictEqual([]);
  });
});
