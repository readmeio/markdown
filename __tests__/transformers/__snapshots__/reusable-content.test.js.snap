// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reusable content transfomer should replace a reusable content block if the block is provided 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Test",
        },
      ],
      "data": Object {
        "hProperties": Object {
          "id": "test",
        },
        "id": "test",
      },
      "depth": 1,
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "link",
            },
          ],
          "title": null,
          "type": "link",
          "url": "http://example.com",
        },
      ],
      "type": "paragraph",
    },
  ],
  "tag": "Test",
  "type": "reusable-content",
}
`;

exports[`reusable content transfomer should replace a reusable content block with multiple words if the block is provided 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Test",
        },
      ],
      "data": Object {
        "hProperties": Object {
          "id": "test",
        },
        "id": "test",
      },
      "depth": 1,
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "link",
            },
          ],
          "title": null,
          "type": "link",
          "url": "http://example.com",
        },
      ],
      "type": "paragraph",
    },
  ],
  "tag": "MyCustomComponent",
  "type": "reusable-content",
}
`;
